### PR TITLE
Asher dl

### PR DESCRIPTION
**Pull request by AsherDLL**
Adding a dashboard for permission changes in the users. It can be accessed through http://.../permission 

TBC: Make that only root and administrative users can access to this dashboard.
TBD: Add a button to ban a user.

Adding new files in documentation/database folder, which include diagrams of
Laravel structure. Diagrams will be joined as a single file in a further
commit.

### CODE REVIEW

File: app/User.php [hunk 1]
```diff
@@ -35,11 +35,19 @@ public function sharedFiles(){
         return $this->hasMany('App\Share', 'idUser');
     }
 
+
+     public function permission(){
+        return $this->hasOne('App\UserType', 'id', 'idUserType');
```
Comment (rikyfocil) [inline app/User.php:40]: Should be belongs according to Laravel standards.

 UserType may be a better name but you can leave it as permission

File: app/User.php [hunk 2]
```diff
@@ -35,11 +35,19 @@ public function sharedFiles(){
...
         return $this->hasMany('App\Member', 'idMember');
     }
 
     public function ownedGroups(){
         return $this->hasMany('App\Group', 'idOwner');
+
     }
-}
+
+}
```
Comment (rikyfocil) [inline app/User.php:53]: Add blank line at end of file

File: resources/views/permissions.blade.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
+
+@extends ('layouts.main')
+	@section('content')
+
+	<table class="table">
```
Comment (rikyfocil) [inline resources/views/permissions.blade.php:5]: Make table like the rest of them. You can base on the files one

File: resources/views/permissions.blade.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+			<th>User</th>
+			<th>Mail</th>
+			<th>Permission</th>
+			<th>Save changes</th>
+		</thead>
+		@foreach($users as $user)
+			<tbody>
+				<td>{{$user->name}}</td>
+				<td>{{$user->email}}</td>
+				{{ Form::open(array('url' => 'permission/' . $user->id)) }}
```
Comment (rikyfocil) [inline resources/views/permissions.blade.php:16]: Use route and pass user->id as param

File: resources/views/permissions.blade.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+				<td>
+					<select name ="permission">
+						<option {{ $user->permission->id == 1? "selected" :""}} value="1">Standard</option>
```
Comment (rikyfocil) [inline resources/views/permissions.blade.php:19]: Should not be hardcoded. You may pass the user types available from the controllers

File: resources/views/permissions.blade.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+				</td>
+				<td>				
+                    {{ Form::hidden('_method', 'PUT') }}
+                    {{ Form::submit('Change') }}
+                {{ Form::close() }}
+				</td>
+			</tbody>
+		@endforeach
+	</table>
+@endsection
```
Comment (rikyfocil) [inline resources/views/permissions.blade.php:33]: End of line here

File: routes/web.php [hunk 7]
```diff
@@ -11,6 +11,9 @@
 |
 */
 
+use App\User;
+use Illuminate\Http\Request;
```
Comment (rikyfocil) [inline routes/web.php:15]: Remove this as we are moving all logic to controllers

File: routes/web.php [hunk 8]
```diff
@@ -24,6 +27,18 @@
 Route::get('/users_complete', 'UsersController@usersLike');
 Route::get('/users_parcial', 'UsersController@usersOnlyLike');
 
+Route::get('permission',function () {
```
Comment (rikyfocil) [inline routes/web.php:30]: Move to a controller

File: resources/views/permissions.blade.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (rikyfocil) [inline resources/views/permissions.blade.php:1]: Move the file to a folder to keep things organized

File: routes/web.php [hunk 10]
```diff
@@ -24,6 +27,18 @@
...
+    return view('permissions', ['users' => User::with("permission"
+)->get()]);
+});
+
+Route::put('permission/{id}',function (Request $request, $id) {
+	$change_permission = App\User::find($id);
```
Comment (rikyfocil) [inline routes/web.php:36]: Should validate for null or use firstOrFail()

File: routes/web.php [hunk 11]
```diff
@@ -24,6 +27,18 @@
...
+	$change_permission->idUserType = $request["permission"];
```
Comment (rikyfocil) [inline routes/web.php:37]: Should validate permission to be present and valid

File: routes/web.php [hunk 12]
```diff
@@ -24,6 +27,18 @@
...
+	$change_permission->save();
+	return Redirect::to("permission");
```
Comment (rikyfocil) [inline routes/web.php:39]: we use return redirect()->route("name") in the rest of controllers. Can you change it to be consistent?

**Review by rikyfocil — CHANGES_REQUESTED** [review]
Also to the review:

Please add the validation for only allowing root users to change this

Please remove all diagrams as we may not include files that we don´t own. (And they are in the wrong folder)

There is no way to arrive to this view. Please add one, only for root users.

Also please remove virtual users from the list

File: app/User.php (hunk 2 above)
Comment (AsherDLL) [reply app/User.php:53]: Fixed

File: resources/views/permissions.blade.php (hunk 9 above)
Comment (AsherDLL) [reply resources/views/permissions.blade.php:1]: Do you want me to create a new folder named "permissions"? or where do you want me to move the file?

File: resources/views/permissions.blade.php (hunk 6 above)
Comment (AsherDLL) [reply resources/views/permissions.blade.php:33]: Fixed

File: resources/views/permissions.blade.php (hunk 9 above)
Comment (AsherDLL) [reply resources/views/permissions.blade.php:1]: Fixed

File: resources/views/permissions.blade.php (hunk 3 above)
Comment (AsherDLL) [reply resources/views/permissions.blade.php:5]: Fixed

File: routes/web.php (hunk 7 above)
Comment (AsherDLL) [reply routes/web.php:15]: Fixed

File: routes/web.php (hunk 8 above)
Comment (AsherDLL) [reply routes/web.php:30]: Fixed

File: routes/web.php (hunk 10 above)
Comment (AsherDLL) [reply routes/web.php:36]: Fixed

File: routes/web.php (hunk 11 above)
Comment (AsherDLL) [reply routes/web.php:37]: Fixed

File: routes/web.php (hunk 12 above)
Comment (AsherDLL) [reply routes/web.php:39]: Fixed

File: app/User.php (hunk 1 above)
Comment (AsherDLL) [reply app/User.php:40]: Won't fix

File: resources/views/permissions.blade.php (hunk 4 above)
Comment (AsherDLL) [inline resources/views/permissions.blade.php]: Fixed

File: resources/views/permissions.blade.php (hunk 5 above)
Comment (AsherDLL) [reply resources/views/permissions.blade.php:19]: Fixed

File: resources/views/permissions.blade.php (hunk 4 above)
Comment (AsherDLL) [reply resources/views/permissions.blade.php:16]: Won't Fix

**Pull request closed without merging**